### ISSUE TITLE
Sieve - can't make it work

### ISSUE THREAD

**Issue by europacafe** [cypht-org/cypht#1153]
## 💬 Question
Mail server: stalwart + snappymail webmail ---> tls sieve work (mail.mydomain.com, ssl/tls, 4190)
Cypht 2.1.0 + stalwart --> tls sieve setting does not work, can't save server config

When opening a mail content, it always reports "Sieve: invalid config host for ....my mailbox name.."
![cypht_sieve_invalid_config](https://github.com/user-attachments/assets/72f28639-d098-4a71-bdf1-4c43d28a3481)

On stalwart mail server, below is the sieve config
![stalwart_sieve_tls_config](https://github.com/user-attachments/assets/d91123bb-190c-4133-88b8-2287a073c628)

On snappymail webmail, the sieive config works
![snappymail_sieve_config](https://github.com/user-attachments/assets/320d895f-c9db-42e0-bfd2-0ccbff6d0d26)

On cypht env file, below is the config:
![cypht_sieve_env_config](https://github.com/user-attachments/assets/0821f0ce-0f2a-488b-b857-ecab47ac9de4)

When I tried to enable sieve on one of my IMAP accounts as below, after saving the config, the screen froze for over 10 secs and came back with an offline label and it didn't save.
![image](https://github.com/user-attachments/assets/7113de0d-a7c1-4191-a442-8a6c8d378171)
I'm not sure what I did wrong. (I enable sieve on Settings/Site/Enable sieve filter)

**Comment by josaphatim** [comment]
@europacafe thank you for reporting this. What if you remove the protocol in sieve host input `mail.brhofen.com:4190` ?

**Comment by europacafe** [comment]
I did try that too. Not working as well.

**Comment by marclaporte** [comment]
Please try Cypht master

**Comment by europacafe** [comment]
How? (with docker compose)

**Comment by marclaporte** [comment]
> How? (with docker compose)

@jonocodes This is not yet coded, right?

**Comment by marclaporte** [comment]
Please try this: https://github.com/cypht-org/cypht/pull/1135/files (just manually modify the file)

**Comment by europacafe** [comment]
just changed to `$server = Hm_IMAP_List::get($imap_server_id, true);`, restarted cypht container
same behavior for both tls://mail.mydomain.com:4190 and mail.mydomain.com:4190

**Comment by Shadow243** [comment]
@europacafe have you checked that the server firewall gives access to 4190 on your server ?

**Comment by europacafe** [comment]
I did open 4190 port on my pfSense. Here is the test result
![image](https://github.com/user-attachments/assets/fd48165f-9a79-4cf1-b634-147e7e3ab6c1)

As mentioned earlier, my snappymail webmail can connect to sieve.

**Comment by jonocodes** [comment]
Yes, I would think you could use the docker compose setup to test this.
Also check the cypht service logs to see if there are any errors.

**Comment by europacafe** [comment]
> Yes, I would think you could use the docker compose setup to test this. Also check the cypht service logs to see if there are any errors.

My cypht is always running as a container, and the suggested openssl test was carried out inside cypht container.

**Comment by europacafe** [comment]
With docker image v2.2.0, this is still an issue

**Comment by jonocodes** [comment]
Do we know that this works as expected when not using docker (using php locally without any container)?

**Comment by marclaporte** [comment]
Is still an issue with Cypht 2.4.0?
https://hub.docker.com/r/cypht/cypht

**Comment by europacafe** [comment]
Yes.

**Comment by Shadow243** [comment]
Will be solved here https://github.com/cypht-org/cypht/pull/1199

**Comment by Shadow243** [comment]
@europacafe please confirm on your side.

**Comment by europacafe** [comment]
With cypht 2.4.0 docker container, it is still not working at all, with Stalwart mail server.

Fyi. With snappymail web client, Sieve works only with ssltls, but not starttls. They are working on the issue.

**Comment by Shadow243** [comment]
> With cypht 2.4.0 docker container

2.4.0 is not yet compatible; it will work with the master branch. Full compatibility will come with version 2.5.0, not 2.4.0.

**Comment by europacafe** [comment]
Great. I may build it from the master and try it. Will feedback later.

**Comment by cptsanifair** [comment]
@europacafe u can try to connect with this tool https://github.com/thsmi/sieve to test the connection and sieve.

**Comment by marclaporte** [comment]
> I may build it from the master and try it. Will feedback later.

@europacafe we need you :-)

**Comment by marclaporte** [comment]
@europacafe @jonocodes @cptsanifair 
The monthly Cypht community meetings are a great opportunity to show bugs: https://github.com/cypht-org/cypht/wiki/Monthly-Community-Meetings
